scipy/signal/spectral.py

```diff
@@ -511,11 +512,481 @@ def spectrogram(x, fs=1.0, window=('tukey',.25), nperseg=256, noverlap=None,
+    freqs, time, Sxx = _spectral_helper(x, x, fs, window, nperseg, noverlap,
                                         nfft, detrend, return_onesided, scaling,
-                                        axis, mode=mode)
+                                        axis, mode='psd')
+
+    return freqs, time, Sxx
+
+
+def check_COLA(window, nperseg, noverlap, tol=1e-10):
+    """
+    Check whether the Constant OverLap Add (COLA) constraint is met
+
+    Parameters
+    ----------
+    window : str or tuple or array_like
+        Desired window to use. See `get_window` for a list of windows and
+        required parameters. If `window` is array_like it will be used
+        directly as the window and its length must match nperseg.
+    nperseg : int
+        Length of each segment.
+    noverlap : int
+        Number of points to overlap between segments.
+    tol : float, optional
+        The allowed variance of a bin's weighted sum from the median bin
+        sum.
+
+    Returns
+    -------
+    verdict : bool
+        `True` if chosen combination satisfies COLA within `tol`, `False`
+        otherwise
+
+    See Also
+    --------
+    stft: Short Time Fourier Transform
+    istft: Inverse Short Time Fourier Transform
+
+    Notes
+    -----
+    In order to enable inversion of an STFT via the inverse STFT in
+    `istft`, the signal windowing must obey the constraint of "Constant
+    OverLap Add" (COLA). This ensures that every point in the input data is
+    equally weighted, thereby avoiding aliasing and allowing full
+    reconstruction.
+
+    Some examples of windows that satisfy COLA:
+        - Rectangular window at 0% & 50% overlap
+        - Bartlett window at 50% overlap
+        - Hann window at 50% & 75% overlap
+        - Any Blackman family window at 2/3 overlap
+        - Any window with ``noverlap = nperseg-1``
+
+    A very comprehensive of other windows may be found in [2]_, wherein the
+    COLA condition is satisfied when the "Amplitude Flatness" is unity.
+
+    .. versionadded:: 0.18.0
+
+    References
+    ----------
+    .. [1] Julius O. Smith III, "Spectral Audio Signal Processing", W3K
+           Publishing, 2011,ISBN 978-0-9745607-3-1.
+    .. [2] G. Heinzel, A. Ruediger and R. Schilling, "Spectrum and spectral
+           density estimation by the Discrete Fourier transform (DFT),
+           including a comprehensive list of window functions and some new
+           at-top windows", 2002,
+           http://hdl.handle.net/11858/00-001M-0000-0013-557A-5
+    """
+
+    nperseg = int(nperseg)
+    if nperseg < 1:
+        raise ValueError('nperseg must be a positive integer')
+
+    if noverlap >= nperseg:
+        raise ValueError('noverlap must be less than nperseg.')
+    noverlap = int(noverlap)
+
+    if isinstance(window, string_types) or type(window) is tuple:
+        win = get_window(window, nperseg)
+    else:
+        win = np.asarray(window)
```

Comment:
See also : https://github.com/scipy/scipy/issues/5744#issuecomment-173245564

> Perhaps a better approach would be to add to get_window so that passing a function or an array to it would also return reasonable results.